ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Simple.hs

```diff
@@ -20,3 +27,17 @@ data TriggerHardFork =
     -- | Never trigger a hard fork
   | TriggerHardForkNever
   deriving (Show, Generic, NoThunks)
+
+instance SerialiseNodeToClient blk TriggerHardFork where
```

Comment:
My initial thought is that this would also belong in the `Ouroboros.Consensus.HardFork.Combinator.Serialisation.SerialiseNodeToClient` module.

But if I recall correctly, I think you had asked why that module of orphans exists. I don't have a good answer (yet?).

Reply:
Orphan instances are a consistent source of annoyance to me. I'd prefer to avoid them where possible.

Reply:
I notice often e.g. [here](https://github.com/input-output-hk/ouroboros-network/pull/3214#discussion_r668905769), that existing orphans encourage new orphans.

Here is my explanation of the mechanism. We have an "orphans" module, `O`, that imports many modules/types and hence will be near the top of the module dependency tree. When adding a new instance in module `M` that requires one of those orphan instances, it's likely that importing `O` from `A` will cause a module cycle. This cycle could be resolved by moving the relevant instances out of `O` to their respective types (i.e. making them non-orphan instances), but the instances often depend on each other and hence would require even more instance be moved out of `O`. We often just pick the easy option and move our new instance from `M` to `O` as an orphan instance. Perhaps there was some good reason to start this off in the first place, but I wouldn't be surprised if most instances are orphans do to this mechanism.